src/hooks/useLockHistory.ts

```diff
@@ -1,22 +1,77 @@
-import { LockHistory } from '@/utils/boost'
+import { useAddress } from './useAddress'
+import useSWRInfinite from 'swr/infinite'
+import { useMemo } from 'react'
+import { toDaysSinceStart } from '@/utils/date'
+import { useChainId } from './useChainId'
+import { CHAIN_START_TIMESTAMPS } from '@/config/constants'
 
-const FAKE_LOCKS = [
-  {
-    day: 0,
-    amount: 1000,
-  },
-  {
-    day: 10,
-    amount: 5000,
-  },
-  {
-    day: 20,
-    amount: 4000,
-  },
-]
+type LockingHistoryEntry =
+  | {
+      eventType: 'LOCKED'
+      executionDate: string
+      transactionHash: string
+      holder: string
+      amount: string
+      logIndex: string
+    }
+  | {
+      eventType: 'UNLOCKED'
+      executionDate: string
+      transactionHash: string
+      holder: string
+      amount: string
+      logIndex: string
+      unlockIndex: string
```

Comment:
Out of curiosity: What do the `logIndex` and `unlockIndex` store and why is there no `lockIndex`?

Reply:
`logIndex` is something technical and unrelated to locking. It is the index of the emitted event: https://sepolia.etherscan.io/tx/0xa7209be9aa51f929bc3f4506bf0a12b8e0005188c4207ef9af2dfbb9f4194f3d#eventlog (Here index 63 is a Lock event for instance)

`unlockIndex`: The index of the unlock action. Each unlock index will be assigned a timestamp to check if it can already be withdrawn. 
That's also why there is no lockIndex, the unlockIndex is needed to check if a unlock can already be withdrawn or not. (24 hours after unlocking, the tokens can be withdrawn)